detectron2/config/defaults.py

```diff
@@ -531,6 +531,10 @@
 # Save a checkpoint after every this number of iterations
 _C.SOLVER.CHECKPOINT_PERIOD = 5000
 
+# Used with `hooks.BestCheckpointer`. Checkpoints every best model based on given eval metric.
+_C.SOLVER.BEST_CHECKPOINTER = CN({"ENABLED": False})
+_C.SOLVER.BEST_CHECKPOINTER.METRIC = "bbox/AP50"
+
```

Comment:
please remove new config options as we expect users to add the hook to their training script directly